internal/dag/ingress_processor.go

```diff
@@ -208,7 +215,20 @@ func route(ingress *networking_v1.Ingress, path string, service *Service, client
 		return r, nil
 	}
 
-	r.PathMatchCondition = &PrefixMatchCondition{Prefix: path}
+	switch pathType {
+	case networking_v1.PathTypePrefix:
+		if path == "/" {
+			r.PathMatchCondition = &PrefixMatchCondition{Prefix: path}
+		} else {
+			prefixRegex := regexp.QuoteMeta(strings.TrimRight(path, "/")) + prefixPathEnsureSegmentOnlyMatchRegex
+			r.PathMatchCondition = &RegexMatchCondition{Regex: prefixRegex}
```

Comment:
A refactor I am considering: this type of path match would still be `PrefixMatchCondition` but the type would add an field bool option as to whether we do a string prefix match or path segment match only. We would then move this logic to a lower level when we convert to the Envoy config type instead, keeping the DAG simpler. We would do the regex conversion if the segment match is desired.

I didn't make this the default behavior b/c we may not want the segment match only for HTTPProxy etc. but if we do, it makes this simpler still, and the above refactor is even better.

Reply:
Also this refactor makes it so the tests for the DAG don't care how the prefix is implemented which is always good

Reply:
For any reviewers this is the main thread to look at ^^

Summary: Prefix path matches in Ingress are not "string prefix" matches but path segment matches (prefix of `/foo` matches `/foo/bar` but not `/foobar`)
- Do we want to make this the case for other resources as well?
- The comments above describe a possible refactor from this current implementation

Reply:
from an in-person chat ImplementationSpecific should be like our current rules:
- move the "looks like regex" handling to this type
- if not a regex, you get a "string prefix" match like we support today